src/Widgets/calendar/Util.vala

```diff
@@ -150,6 +160,36 @@ namespace Util {
         return datetime.add_full (0, 0, 0, -datetime.get_hour (), -datetime.get_minute (), -datetime.get_second ());
     }
 
+    public string get_style_calendar_color (E.SourceCalendar cal) {
+        var uid = cal.source.uid;
+        var color = cal.dup_color ();
+        string css_class = "event-color-%s".printf (uid);
+
+        string style = """
+                        .%s {
+                            background-color: alpha(%s, 0.15);
+                            color: shade(%s, 0.65);
+                            border-radius: 4px;
+                            box-shadow: 0 1px 2px alpha (#000, 0.2),
+                                        0 1px 3px alpha (%s, 0.15);
+                        }
+                        .%s image {
+                            color: shade(%s, 0.65);
+                        }
+                       """.printf(css_class, color, color, color, css_class, color);
```

Comment:
Instead of inserting `color` 4 times, make a variable (using `@define-color`) and only insert it once.